### PR TITLE
Feature/pause menu

### PR DESCRIPTION
**Pull request by Lucas4185**

### CODE REVIEW

**Comment by SjorsMaster** [comment]
Can't be seen through walls but seems to work.

File: Assets/SteamVR/Extras/SteamVR_LaserPointer.cs [hunk 1]
```diff
@@ -91,11 +91,13 @@ public virtual void OnPointerOut(PointerEventArgs e)
 
         private void Update()
         {
-            if (!isActive)
+            if (active != isActive)
```
Comment (SjorsMaster) [inline Assets/SteamVR/Extras/SteamVR_LaserPointer.cs:94]: Consider renaming active to state for a more logic naming.

File: Assets/Scripts/UI/PauseMenu/VRPointer.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
+using UnityEngine;
+using UnityEngine.UI;
+using Valve.VR.Extras;
+
+namespace LekAanDek
```
Comment (SjorsMaster) [inline Assets/Scripts/UI/PauseMenu/VRPointer.cs:5]: select a proper namespace

**Review by SjorsMaster — CHANGES_REQUESTED** [review]
Please see feedback on scripts

File: Assets/SteamVR/Extras/SteamVR_LaserPointer.cs (hunk 1 above)
Comment (Lucas4185) [reply Assets/SteamVR/Extras/SteamVR_LaserPointer.cs:94]: Dit script is niet van mij is van Valve

File: Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    [ExecuteInEditMode] //Disable if you don't care about previewing outside of play mode
+    public class WorldSpaceOverlayUI : MonoBehaviour
+    {
+        private const string _shaderTestMode = "unity_GUIZTestMode"; //The magic property we need to set
+
+        [SerializeField]
+        public UnityEngine.Rendering.CompareFunction desiredUIComparison = UnityEngine.Rendering.CompareFunction.Always; //If you want to try out other effects
+
+        [Tooltip("Set to blank to automatically populate from the child UI elements")]
+        [SerializeField]
```
Comment (SjorsMaster) [inline Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs:20]: Why are you serializing public variables? Either make them private, or remove serialization.

File: Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        public Graphic[] uiElementsToApplyTo;
+        //Allows us to reuse materials
+        private Dictionary<Material, Material> materialMappings = new Dictionary<Material, Material>();
```
Comment (SjorsMaster) [inline Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs:23]: Please add prefix to the private member

File: Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            if (uiElementsToApplyTo.Length == 0)
+            {
+                uiElementsToApplyTo = gameObject.GetComponentsInChildren<Graphic>();
+            }
+            foreach (var graphic in uiElementsToApplyTo)
+            {
+                Material material = graphic.materialForRendering;
+                if (material == null)
+                {
+                    Debug.LogError($"{nameof(WorldSpaceOverlayUI)}: skipping target without material {graphic.name}.{graphic.GetType().Name}");
```
Comment (SjorsMaster) [inline Assets/Scripts/UI/PauseMenu/WorldSpaceOverlayUI.cs:35]: Please remove any debug logs

**Review by SjorsMaster — APPROVED** [review]
Works fine, please consider fixing the pointer layer in the next sprint as part of the polishing

**Pull request merged**